src/wire/ipv6ext.rs

```diff
@@ -0,0 +1,246 @@
+use {Error, Result};
+
+enum_with_unknown! {
+    /// IPv6 Extension Header Option type
+    pub doc enum OptionType(u8) {
+        /// Pad1 option
+        Pad1 =  0,
+        /// PadN option
+        PadN =  1
+    }
+}
+
+/// A read/write wrapper around a variable number of IPv6 Extension Header Options
+#[derive(Debug, PartialEq)]
+pub struct Packet<T: AsRef<[u8]>> {
+    buffer: T
+}
+
+// Format of Option
+//
+// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+- - - - - - - - -
+// |  Option Type  |  Opt Data Len |  Option Data
+// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+- - - - - - - - -
+//
+//
+// See https://tools.ietf.org/html/rfc8200#section-4.2 for details.
+mod field {
+    use wire::field::*;
+
+    // 8-bit identifier of the type of option
+    pub const TYPE:     usize = 0;
+    // 8-bit unsigned integer. Length of the DATA field of this option, in octets
+    pub const LENGTH:   usize = 1;
+	// Variable-length field. Option-Type-specific data.
+    pub const DATA:     Rest  = 2..;
+}
+
+impl<T: AsRef<[u8]>> Packet<T> {
+    /// Create a raw octet buffer with an IPv6 Extension Header Option packet structure.
+    pub fn new(buffer: T) -> Packet<T> {
+        Packet { buffer }
+    }
+
+    /// Consume the packet, returning the underlying buffer.
+    pub fn into_inner(self) -> T {
+        self.buffer
+    }
+
+    /// Return the option type.
+    #[inline]
+    pub fn option_type(&self) -> OptionType {
+        let data = self.buffer.as_ref();
+        OptionType::from(data[field::TYPE])
+    }
+
+    /// Return the length of the data.
+    ///
+    /// # Panics
+    /// The function panics if the type does not support this field
+    #[inline]
+    pub fn option_data_length(&self) -> u8 {
+        let data = self.buffer.as_ref();
+        data[field::LENGTH]
+    }
+
+    /// Return the option data.
+    ///
+    /// # Panics
+    /// The function panics if the type does not support this field
+    #[inline]
+    pub fn option_data(&self) -> &[u8] {
+        let data = self.buffer.as_ref();
+        let len = (self.option_data_length() as usize) + field::DATA.start;
+        &data[field::DATA.start..len]
+    }
+}
+
+impl<T: AsRef<[u8]> + AsMut<[u8]>> Packet<T> {
+    /// Set the option type.
+    #[inline]
+    pub fn set_option_type(&mut self, value: OptionType) {
+        let data = self.buffer.as_mut();
+        data[field::TYPE] = value.into();
+    }
+
+    /// Set the option data and data length
+    ///
+    /// # Panics
+    /// The function panics if the type does not support the length or data fields
+    #[inline]
+    pub fn set_option_data(&mut self, value: &[u8], len: u8) {
```

Comment:
`payload_mut` or `data_mut`. See some of the other implementations for details.

Reply:
modified `option_data` to be `data` and conformed to the examples.